_posts/2020-10-02-nlp-translation.md

```diff
@@ -0,0 +1,205 @@
+
+
+## Training
+
+### Loss Function
+For both of the models, we use Negative Log-Likelihood loss with label smoothing. 
+The models output a probability for each word of the vocabulary for the translated sentence.
+From this, we can compute $$NLLLoss(\mathbf{\hat{y}}, \mathbf{y})$$ where $$\mathbf{\hat{y}}$$ is the model output and $\mathbf{y}$ is the target. 
+
+
+$$ Smooth Loss = -mean (log softmax(\mathbf{\hat{y}})) $$
+$$ Loss = confidence * NLLLoss + smoothing * SmoothLoss $$
+
+Where $$confidence = 1 - smoothing$$. The smoothing is set to a value of 0.1 for both tasks.
+
+### Optimization
+As we have seen above, both models have a very high number of parameters to train. This can be an issue when using GPUs, as the model needs to fit in memory.
+Also, back-propagation requires the memory to be at least twice the size of the model for it to work in memory. For that, instead of using regular precision, we used
+mixed-precision training, where most computations are done in `Float16`. We use a centralized version of `Adam`, where gradients are aggregated amongst all workers before 
+updating weights.
+
+### Datasets
+Both tasks use the same test set, but are trained on slightly different data sets:
+- The LSTM is trained on the English to German World Machine Translation 16 (WMT16) dataset, comprising of 3,975,116 translated sentences
+- The Transformer is trained on the English to German World Machine Translation 17 (WMT17) dataset, comprising of 4,590,101.
+ 
+
+More details on both tasks can be found on our [documentation](https://mlbench.readthedocs.io/en/latest/benchmark-tasks.html#task-4-machine-translation).
+
+## Results
+
+Let us now get to fun part; the results. As previously discussed, those models have important training times, and the aim of MLBench is to study the benefit of distribution.
+For reproducibility purposes, here is the hardware and software we have used:
+- Cloud service: Google Cloud
+- Machine Type: `n1-standard-4`
+- PyTorch 1.5.1
+- NVIDIA Tesla-T4 GPU (1 per node)
+- 4 cores and 15GB of RAM
+- NCCL communication backend 
+
+The goal for both models is determined by the Bilingual Evaluation Understudy Score (BLEU):
+- The LSTM task stops when reaching a BLEU Score of 24.0
+- The Transformer task stops when reaching a BLEU score of 25.0
+
+The models are trained on 1,2,4,8 and 16 workers, and all step times are precisely measured to obtain an accurate speed up quantification.
+Speedups are computed with respect to the 1 worker case, and are intended to illustrate the distributive capabilities of the task.
+
+### Overall Speedups
+
+The graphs below show the time speedups for the LSTM model and Transformer model (respectively). 
+
+<a href="{{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_speedup.png" data-lightbox="task4a_speedups" data-title="Speedups for GNMT">
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_speedup.png)
+  *GNMT Speedups*
```

Comment:
With the spacing on the blog post, the label "GNMT Speedups" is visually closer to the graphs below than above. Maybe you can add some spacing below, or move the labels to above the plots.